plugins/processors/noise/noise.go

```diff
@@ -0,0 +1,136 @@
+package noise
+
+import (
+	"math"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+const (
+	defaultSensitivity = 1.0
+	defaultEpsilon     = 1.0
+)
+
+var sampleConfig = `
+  [[processors.noise]]
+    sensitivity = 1.0
+    epsilon = 1.0
+    ignore_fields = []
+    ignore_measurements = []
+`
+
+var laplaceValue = 0.0
+var fieldExcludeSet = make(map[string]bool)
+var measurementExcludeSet = make(map[string]bool)
+
+type Noise struct {
+	Sensitivity        float64
+	Epsilon            float64
+	IgnoreFields       []string        `toml:"ignore_fields"`
+	IgnoreMeasurements []string        `toml:"ignore_measurements"`
+	Log                telegraf.Logger `toml:"-"`
+}
+
+func (p *Noise) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Noise) Description() string {
+	return "Generates noise based on the Laplace distribution and add that to " +
+		"all numerical field values. Exclusions can be listed in the config file."
+}
+
+// Gets a random, positive Integer from a Laplace distribution and adds it to a
+// given value
+func (p *Noise) addLaplaceUInt64(value uint64) uint64 {
+	return value + value*uint64(math.Round(math.Abs(p.getRandomLaplaceNoise())))
+}
+
+// Gets a random Integer from a Laplace distribution and adds it to a
+// given value
+func (p *Noise) addLaplaceInt64(value int64) int64 {
+	return value + value*int64(math.Round(math.Abs(p.getRandomLaplaceNoise())))
+}
+
+// Gets a random, positive float from a Laplace distribution and adds it to a
+// given value
+func (p *Noise) addLaplaceFloat64(value float64) float64 {
+	return value + value*math.Abs(p.getRandomLaplaceNoise())
+}
+
+// Returns a random float from a Laplace distribution. If nil is passed, a
+// random seed will be generated internally
+func (p *Noise) getRandomLaplaceNoise() float64 {
+	l := distuv.Laplace{
+		Mu:    0,
+		Scale: p.Sensitivity / p.Epsilon,
+		Src:   nil,
+	}
+	laplaceValue = l.Rand()
+	return laplaceValue
+}
+
+// Takes a value as interface and adds laplace noise to any given numerical type
+func (p *Noise) addNoiseToValue(value interface{}) interface{} {
+	switch v := value.(type) {
+	case int64:
+		value = p.addLaplaceInt64(value.(int64))
```

Comment:
Please use the typed field instead
```suggestion
		value = p.addLaplaceInt64(v)
```
Same for the other "cases".

Reply:
Implemented with 99eb43507214d0e36054cce7c88a8540c54d1210